web/src/features/charts/ProductionSourceLegendList.tsx

```diff
@@ -10,10 +13,59 @@ export default function ProductionSourceLegendList({
   sources: ElectricityModeType[];
   className?: string;
 }) {
+  const isMobile = useIsMobile();
+
+  return (
+    <TooltipWrapper
+      tooltipContent={<ProductionSourceTooltip sources={sources} isMobile={isMobile} />}
+      tooltipClassName={
+        isMobile
+          ? 'bg-transparent shadow-none border-transparent dark:bg-transparent items-center flex flex-col p-0 mb-2'
+          : 'rounded-2xl min-w-44 dark:bg-gray-900/80 dark:border-1 dark:border-gray-700 mx-5 backdrop-blur bg-white/80 mb-2'
+      }
+      side="bottom"
+      isMobile={isMobile}
+    >
+      <div className={twMerge('flex w-fit flex-row gap-1.5 py-1', className)}>
+        {sources.map((source, index) => (
+          <ProductionSourceLegend key={index} electricityType={source} />
+        ))}
```

Comment:
As the bot suggested you can use the source object directly as a key here as well just as you did later in the code.